### ISSUE TITLE
How to provide more custom property values to resolvers?

### ISSUE THREAD

**Issue by pekkah** [graphql-dotnet/graphql-dotnet#637]
Current solution is to set UserContext but you can only have one.

I propose adding a property bag into the options and pass it along until it reaches the resolvers.  Provide simple Set(key, value), Get<T>(key) for manipulation. This requirement is coming from the server side as currently I'm hijacking the UserContext to pass the MessageHandlingContext to resolvers so I can access it there.

**Comment by PascalSenn** [comment]
as in #648 mentioned I like the idea of a property bag. 
It'd be nice to pass it down as an object in resolvers. It'd also be nice if it's accessible inside the type tree. 
Example: 
QueryType -> FooType -> BarType -> FooBarType -> StringType
  {   }           ->   {    }       -> { f: "B" }  ->  { f: "B" }      -> { f: "B" }   

That'd be a very handy way to pass down scopes of a request, specified as arguments. 
```graphql
query Sample($f: String!){
    foo {
        bar(f: $f) {
            fooBar {
                name
            }
        }
    }
}
```

**Comment by joemcbride** [comment]
@PascalSenn You can access variables on `ResolveFieldContext` today if you need that scenario.

I'm seeing this as a global available to all resolvers like UserContext.  So you could add to it in a Resolver if you would like, but it wouldn't be node specific.

**Comment by PascalSenn** [comment]
@joemcbride  
I'd like to revive this. 

What about this scenario. Our API is scoped. We have an array of customers that we'd like to resolve. 
Underneath the customer node we have a fairly deep graphql tree. 
As far as i know the only way to pass down properties is to drill them down the tree.
```csharp
           FooBarModel  Customer(string customer).getFooBar();
                    ->  FooBarModelDeep FooBarModel(string customer).GetFooBarModelDeep();
```
That leads to unnecessary models and a higher complexity over all. 

```graphql
query Sample($f: [String]){
    foo {
        customers(f: $f) {
            fooBar {
               fooBarDeep {
                      name
                   }
            }
        }
    }
}
``` 

input ["FirstEntry", "SecondEntry"]  
result:
```
{
   foo: {
       customers: [
          {
               fooBar: {
                    fooBarDeep: {
                          name:"FirstEntry"
                    }
               }
          },
             {
               fooBar: {
                    fooBarDeep: {
                          name:"SecondEntry"
                    }
               }
          }

        ]
    }
}
```

Is there also already a way around it?

**Comment by SimonCropp** [comment]
fixed as part of #1037
